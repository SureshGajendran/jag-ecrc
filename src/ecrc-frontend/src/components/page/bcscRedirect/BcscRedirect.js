/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState } from "react";
import PropTypes from "prop-types";
import { Redirect } from "react-router-dom";
import axios from "axios";

import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import { Button } from "../../base/button/Button";
import SideCards from "../../composite/sideCards/SideCards";
import {
  isAuthenticated,
  generateJWTToken
} from "../../../modules/AuthenticationHelper";

import "../page.css";
import "./BcscRedirect.css";

export default function BcscRedirect({ page: { header, saveOrg } }) {
  const [bcscUrl, setBcscUrl] = useState("");
  const [toHome, setToHome] = React.useState(false);

  React.useEffect(() => {
    window.scrollTo(0, 0);

    if (!isAuthenticated("tou")) setToHome(true);

    const payload = {
      authorities: ["ROLE"],
      visited: ["orgValidation", "orgVerification", "tou", "bcscRedirect"]
    };
    const token = generateJWTToken(payload);

    axios
      .get(`/ecrc/protected/getBCSCUrl`, {
        headers: {
          Authorization: `Bearer ${token}`
        }
      })
      .then(res => {
        setBcscUrl(res.data);
      })
      .catch(() => {});
  }, []);

  const loginBtn = {
    label: "Login with a BC Services Card",
    buttonStyle: "btn ecrc_btn_login_bcsc",
    buttonSize: "btn",
    type: "button"
  };

  const onLoginClick = () => {
    saveOrg();
    // REDIRECT TO BCSC
    window.open(bcscUrl, "_self");
  };

  if (toHome) {
    return <Redirect to="/" />;
  }

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content col-md-8">
          <h1>Apply for a Criminal Record Check</h1>
          <br />
          <p>
            To apply for a criminal record check online, you must use your{" "}
            <a href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card/login-with-card">
              BC Services Card
            </a>{" "}
            Account. Only cards <b>with a photo</b> are accepted. If it&apos;s
            your first time using your Card to access any online service, you
            need to set up your account for use online by completing a one-time
            security check to{" "}
            <a href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card/login-with-card">
              verify your identity
            </a>
            .
          </p>

          <p>Please select an option below:</p>
          <br />
          <div className="option">
            <div className="row">
              <div
                className="col-lg-6 col-md-12 "
                style={{ marginBottom: "10px" }}
              >
                I have a BC Services Card{" "}
                <span className="underlineText">with a photo</span>
              </div>
              <div className="col-lg-6 col-md-4 alignRight">
                <Button button={loginBtn} onClick={onLoginClick} />
              </div>
            </div>
          </div>

          <div style={{ marginTop: "40px" }}>
            <a href="https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check">
              I do not have a BC Services Card, or I have non-photo BC Services
              Card
            </a>
            .
          </div>
        </div>

        <div className="sidecard">
          <SideCards type={"bcservice"} />
          <SideCards type={"withoutBCServiceCard"} />
        </div>
      </div>
      <Footer />
    </main>
  );
}

BcscRedirect.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }),
    saveOrg: PropTypes.func.isRequired
  }).isRequired
};
