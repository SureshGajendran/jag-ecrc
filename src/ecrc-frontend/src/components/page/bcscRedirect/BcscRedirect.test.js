import React from "react";
import { create } from "react-test-renderer";

import BcscRedirect from "./BcscRedirect";

describe("BcscRedirect Page Component", () => {
  test("Matches the snapshot", () => {
    const header = {
      name: "Criminal Record Check"
    };

    const page = {
      header
    };

    const bcscRedirect = create(<BcscRedirect page={page} />);
    expect(bcscRedirect.toJSON()).toMatchSnapshot();
  });
});
