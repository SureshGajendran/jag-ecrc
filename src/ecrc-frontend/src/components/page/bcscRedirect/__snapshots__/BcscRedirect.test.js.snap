// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BcscRedirect Page Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="https://www2.gov.bc.ca"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div
          className="navbar-brand"
        >
          Criminal Record Check
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <a
              className="nav-item nav-link"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Apply for a Criminal Record Check
      </h1>
      <p>
        Now, that you have confirmed the organization you work or plan to work for, the next step is to complete a criminal record check application.
      </p>
      <p>
        To apply for a criminal record check online, you must use your
         
        <a
          href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card/login-with-card"
        >
          BC Services Card
        </a>
         
        Account.
      </p>
      <p>
        Only cards 
        <b>
          with a photo
        </b>
         are accepted. If it's your first time using your Card to access any online service, you need to set up your account for use online by completing a one-time security check to
         
        <a
          href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card/login-with-card"
        >
          verify your identity
        </a>
      </p>
      <p>
        Please select an option below:
      </p>
      <div
        className="option"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-6 col-md-12 "
            style={
              Object {
                "marginBottom": "10px",
              }
            }
          >
            I have a BC Services Card
             
            <span
              className="underlineText"
            >
              with a photo
            </span>
          </div>
          <div
            className="col-lg-6 col-md-4 alignRight"
          >
            <button
              className="btn ecrc_btn_login_bcsc btn"
              onClick={[Function]}
              type="button"
            >
              Login with a BC Services Card
            </button>
          </div>
        </div>
      </div>
      <div
        className="option"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-8 col-md-12"
            style={
              Object {
                "marginBottom": "10px",
              }
            }
          >
            I 
            <span
              className="underlineText"
            >
              DON’T
            </span>
             have a BC Services Card, or I have one without a photo
          </div>
          <div
            className="col-lg-4 col-md-12 col-sm-12 alignRight"
          >
            <button
              className="btn ecrc_bscs_apply_btn btn"
              onClick={[Function]}
              type="button"
            >
              Find how to apply
            </button>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "marginTop": "40px",
          }
        }
      >
        <button
          className="btn ecrc_accessary_btn btn"
          onClick={[Function]}
          type="submit"
        >
          Cancel and Exit
        </button>
      </div>
    </div>
    <div
      className="sidecard"
    >
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  Get a BC Services Card
                </h2>
                <div
                  className="submit-content"
                >
                  <p>
                    B.C. residents who have lived in the province for at least six months must use a BC Services Card to log in to the online qualification tool. Learn how to get a card.
                    <button
                      className="btn btn-primary dark-button btn btn-md"
                      onClick={[Function]}
                      type="submit"
                    >
                      READ MORE
                    </button>
                  </p>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  Applicants Without a BC Services Card
                </h2>
                <div
                  className="submit-content"
                >
                  <p>
                    Applicants who haven’t lived in B.C.for at least six months or who live outside the province must 
                    <a
                      className="link"
                      href="mailto: temp@temp.com"
                    >
                      email
                    </a>
                     the Liquor and Cannabis Regulation Branch for a Worker Registration Application Form.
                  </p>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
    style={Object {}}
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
